docs/05-reference/api/delete-asset.mdx

```diff
@@ -0,0 +1,18 @@
+---
+title: "DELETE asset"
+---
+
+# `DELETE` asset
+
+`DELETE /asset/{id}` deletes an existing `asset` object.
+
+This will remove an existing asset stored with Livepeer Studio.
+
+>**NOTE:** Assets are not deleted immediately when making this API call. If immediate removal of the asset is required, please [contact support ](https://livepeer.studio/contact)
```

Comment:
Nice :)

Reply:
@victorges @Shih-Yu What if a user stores an asset on IPFS or through our Arweave integration in the future? Assets stored on a decentralized storage providers are immutable and can't be deleted. Should we instead frame this to the developer as archiving and unarchiving assets? The dashboard can then support a view for viewing archived assets in case they want to unarchive.

Reply:
I like the terms archiving/unarchiving for describing the process and should be mentioned for decentralized storage

Reply:
Hmm yeah, I like the idea of calling it archiving when the asset has been stored on a permanent storage like Arweave.

My only concern is having different behaviors when the asset is or isn't stored on such permanent storage. We will likely bill our users based on storage use (with Arweave paid upfront), so when they delete an asset we should also delete it from our internal storage as well as any other non-permanent decentralized storages like IPFS, in which we can unpin the item from our (managed) IPFS nodes. 

On that case, where an asset wasn't on Arweave, we won't be able to "unarchive" to asset anymore, since it will have been deleted from any storage that we held it. Then I don't think it makes sense to "archive" instead of "delete" it in this case.

So WDYT could be the approach on that case? Some ideas:
 - not delete the files from our internal storage to allow every deletion to be undone (expensive for us and maybe limits users)
 - disallow `DELETE`ing an asset **that has been saved on Arweave**, and create a new "archive" API (or option/query on the current DELETE) when we have that, which would only work for assets in permanent storages
 - just have `DELETE` being handled differently, being reversible in one case (saved in permanent storage) or irreversible if not. On this case it might be "just" a naming issue, does it make sense for it to be named "archive" but not be reversible, or "delete" and sometimes reversible?

Reply:
Thanks for laying out these approaches. I'd be in favor of the first approach. In the future I imagine we'll want to default to a decentralized storage option and remove our dependency on google so it's good to prime user expectations around the immutability of decentralized storage providers now. Cold storage sounds like a good way to minimize our costs until we pass through costs to developers.

I can work on introducing an archiving UI in the dashboard.

Reply:
Ideally in the future we default to a decentralized storage provider and give them the option to store redundantly on  google. Then they can delete from google because it will still exist on decentralized storage.